File: setup/src/main/java/es/codeurjc/daw/library/dto/TournamentDTO.java
package es.codeurjc.daw.library.dto;
import java.util.List;

public record TournamentDTO(
    Long id, 
    String name, 
    String type, 
    String status, 
    int maxParticipants, 
    boolean hasImage,
    List<TeamBasicDTO> teams,
    List<MatchBasicDTO> matches
) {}

File: frontend/exam0_minimal/app/dtos/TournamentDTO.ts
export enum TournamentType {
    LIGA = "LIGA",
    ELIMINATORIA = "ELIMINATORIA"
}

export enum TournamentStatus{
    INSCRIPCIONES = "INSCRIPCIONES_ABIERTAS",
    EN_CURSO = "EN_CURSO",
    FINALIZADO = "FINALIZADO"
}

export interface TeamBasicDTO {
    id: number;
    username: string;
    teamName: string;
    hasImage: boolean;
    goals?: number;
    assists?: number;
}

export interface MatchDTO {
    id: number;
    homeTeam: TeamBasicDTO;
    awayTeam: TeamBasicDTO;
    homeGoals: number;
    awayGoals: number;
    played: boolean;
}

export interface TournamentDTO {
    id: number;
    name: string;
    status: TournamentStatus;
    type: TournamentType;
    maxParticipants: number;
    hasImage: boolean;
    teams: TeamBasicDTO[];
    matches: MatchDTO[];
    owner?: { username: string; name: string; };
}
